### PR TITLE
LIME-254: Add Landing Page

### LINKED ISSUE #254: Add Landing Page
https://www.figma.com/file/RRE9g9LIo3a1AK89HBlN6U/BSA_Lime?type=design&node-id=2276-28296&mode=design&t=dMJOmk97NC87WzW1-0

- [ ] add soma anchors for navigation
- [ ] add animation on scroll

### PR DESCRIPTION
**Pull request by stpalamar**
1. Changed route flow. The overview route is the home page for authorized users. 
2. "Download buttons" on the landing page redirect to sign-up since we decided with QA 
3. "Learn more" button has anchor to How It Works section

**UPD**: I have already added a loader when the user is refreshing, the video with route flow was recorded before
**UPD**: Logo for white theme added, screenshots in comments below

**Demo**

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/26843161/15556c54-00d6-4c49-bc74-47a6b43b5124

**Route flow**

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/26843161/00da8d54-94ac-4ad4-bd8e-68e98c2c39aa

### CODE REVIEW

**Review by DavidCharif — DISMISSED** [review]

File: frontend/src/bundles/landing/pages/landing.tsx [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+                        size={ComponentSize.MEDIUM}
+                        label="Download Now and Get Fit"
+                        variant={ButtonVariant.PRIMARY}
+                        className="max-w-[15.5rem]"
+                        onClick={signUpHandler}
+                    />
+                </div>
+                <footer className="container mx-auto flex h-[40rem] flex-col-reverse items-center justify-evenly sm:h-[20rem] md:flex-row md:items-start md:pt-60">
+                    <div>
+                        <img src={logo} alt="Logo" />
```
Comment (valery-chumak) [inline frontend/src/bundles/landing/pages/landing.tsx]: image
The logo in footer dissapears on light theme

File: frontend/src/bundles/landing/pages/landing.tsx [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+
+    return (
+        <main className="text-primary flex h-screen snap-y snap-mandatory flex-col gap-40 overflow-y-auto overflow-x-hidden scroll-smooth">
+            <section
+                className={getValidClassNames(styles.section, styles.container)}
+            >
+                <header className="container mx-auto my-6 flex flex-row items-center justify-between">
+                    <div>
+                        <Link to={AppRoute.ROOT}>
+                            <img src={logo} alt="Logo" />
```
Comment (valery-chumak) [inline frontend/src/bundles/landing/pages/landing.tsx]: Here too
image

**Review by valery-chumak — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/landing/pages/landing.tsx (hunk 2 above)
Comment (valery-chumak) [reply frontend/src/bundles/landing/pages/landing.tsx]: @stpalamar It looks really great, especially the animation) Cool work)

Comment (stpalamar) [reply frontend/src/bundles/landing/pages/landing.tsx]: Thanks! I will fix logo

**Comment by stpalamar** [comment]
Fixed the logo, I took implementation from #265

**Review by valery-chumak — DISMISSED** [review]

**Comment by stpalamar** [comment]
Changed the name of the buttons

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**